tools/add_triage_signature.py

```diff
@@ -713,6 +713,49 @@ def _process_ticket(self, url, issue_key):
         report = self._generate_table_for_report(hosts)
         self._update_triaging_ticket(report)
 
+class MasterFailedToRebootSignature(ErrorSignature):
+    """
+    This signature finds tickets for clusters where at least one master node failed to reboot
+    This signature is only applicable if the host that failed to reboot has bios boot mode
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            comment_identifying_string="h1. Master node failed to reboot",
+            function_impact_label="master_failed_rebooting",
+        )
+
+    def _process_ticket(self, url, issue_key):
+        md = get_metadata_json(url)
+        cluster = md["cluster"]
+        hosts = []
+        for host in cluster["hosts"]:
+            inventory = json.loads(host["inventory"])
+            boot_mode = inventory.get("boot", {}).get("current_boot_mode", "N/A")
+            if (
+                host["role"] == "master"
+                and host["progress"]["current_stage"] == "Rebooting"
+                and host["status"] == "error"
```

Comment:
It's also possible that the host went into error because the cluster went into error, can you prevent that from being a false positive?

Reply:
Yes, I can check:
1. The bootstrap node status is waiting for control plane (in case it's not SNO).
2. If the master is in error than the current_stage must be rebooting
This should be unique enough for the case where the installation failed due to master reboot